src/tp_engine.hpp

```diff
@@ -170,12 +167,23 @@ class TPEngine {
  protected:
   // ** Scoring by Likelihoods
 
+  // Compute the rootward P-PVs for given node.
   void PopulateRootwardLikelihoodPVForNode(const NodeId node_id);
+  // Compute the leafward R-PVs for given node.
   void PopulateLeafwardLikelihoodPVForNode(const NodeId node_id);
+  // Set the P-PVs to match the observed site patterns at the leaves.
   void PopulateLeafLikelihoodPVsWithSitePatterns();
+  // Set the R-PVs to the stationary distribution at the root and rootsplits.
   void PopulateRootLikelihoodPVsWithStationaryDistribution();
-  EdgeId BestEdgeAdjacentToNode(const NodeId node_id, const Direction direction) const;
+  // Find the edge from the highest likelihood tree that is adjacent to given node in
+  // the given direction.
```

Comment:
What actually happens here is in terms of the `tree_source_` data member. Could you be a little more explicit about what's really going on in this method (it's not in terms of likelihood as such).

Reply:
How is this?

```
  // Accomplished by iterating over all adjacent edges using tree_source_ edge map,
  // which gives the best tree id that uses any given edge. Adjacent edge that comes
  // from the best tree is chosen.
  ```

Reply:
That's good, but let's emphasize that the best tree is the one that appears first. Thx!

Reply:
Okay, updated comment to clarify best tree is first tree in collection.